docs/guides/typescript.md

```diff
@@ -399,91 +399,56 @@ A detailed explanation on the slices pattern can be found [here](./slices-patter
 
 If you have some middlewares then replace `StateCreator<MyState, [], [], MySlice>` with `StateCreator<MyState, Mutators, [], MySlice>`. For example, if you are using `devtools` then it will be `StateCreator<MyState, [["zustand/devtools", never]], [], MySlice>`. See the ["Middlewares and their mutators reference"](#middlewares-and-their-mutators-reference) section for a list of all mutators.
 
-### Using a vanilla store as a bound store
-
-Create your vanilla store:
+### A bounded `useStore` hook
 
 ```ts
-import { createStore } from 'zustand/vanilla'
+import { create, useStore } from 'zustand'
 
 interface BearState {
   bears: number
   increase: (by: number) => void
 }
 
-export const initialBearState = { bears: 0 }
-export const vanillaBearStore = createStore<BearState>((set, getState) => ({
-  ...initialBearState,
+const bearStore = create<BearState>()((set) => ({
+  bears: 0,
   increase: (by) => set((state) => ({ bears: state.bears + by })),
 }))
-```
-
-Create a hook to provide a bound store to be used in your component:
 
-```ts
-import { useStore } from 'zustand'
-
-export function useBoundBearStore(): BearState
-export function useBoundBearStore<T>(
-  selector: (state: BearState) => T,
-  equals?: (a: T, b: T) => boolean
-): T
-export function useBoundBearStore(selector?: any, equals?: any) {
-  return useStore(vanillaBearStore, selector, equals)
+function useBearStore(): BearState
+function useBearStore<T>(selector: (state: BearState) => T, equals?: (a: T, b: T) => boolean): T
+function useBearStore<T>(selector?: (state: BearState) => T, equals?: (a: T, b: T) => boolean) {
+  return useStore(bearStore, selector!, equals)
```

Comment:
We need the `!` after `selector` because our types are incorrect, they don't allow writing `useStore({} as StoreApi<unknown>, undefined, () => true)`. I had told @dai-shi about this in #725 but he insisted that passing `undefined` is an incorrect usage. I personally think types should remain true to the implementation otherwise problems like this pop up.

I mean it's not a big deal but still in case if you change your mind and want to make our types more correct then let me know I'll open a PR (we just need to add a `?` after `selector` in out `useStore` type).

Reply:
Yes, I still think this is good. (But, I may change my mind in the future. In that case, I think I know what to do.)